bin/make-release.py

```diff
@@ -0,0 +1,302 @@
+        )
+        return 2
+
+    # Make sure we have all the old git tags
+    subprocess.check_output(
+        f'git pull {upstream_name} master --tags'.split(),
+        stderr=subprocess.STDOUT
+    )
+
+    # We're going to use the last tag to help you write a tag message
+    last_tag, last_tag_message = subprocess.check_output([
+        'git',
+        'for-each-ref',
+        '--sort=-taggerdate',
+        '--count=1',
+        '--format',
+        '%(tag)|%(contents:subject)',
+        'refs/tags'
+    ]).decode('utf-8').strip().split('|', 1)
+
+    commits_since = subprocess.check_output(
+        f'git log {last_tag}..HEAD --oneline'.split()
+    ).decode('utf-8')
+    commit_messages = []
+    for commit in commits_since.splitlines():
+        wo_sha = re.sub('^[a-f0-9]{7} ', '', commit)
+        commit_messages.append(wo_sha)
+
+    print(' NEW CHANGE LOG '.center(80, '='))
+    change_log = []
+    head = '{} ({})'.format(
+        next_version,
+        datetime.datetime.now().strftime('%Y-%m-%d')
+    )
+    head += '\n{}'.format('-' * len(head))
+    change_log.append(head)
+    change_log.extend(['- {}'.format(x) for x in commit_messages])
+    print('\n\n'.join(change_log))
+    print('=' * 80)
+
+    assert commit_messages
+
+    # Edit jobs/setup.py
+    with open('jobs/setup.py') as f:
+        setup_py = f.read()
+    assert "version='{}',".format(current_version) in setup_py
+    setup_py = setup_py.replace(
+        "version='{}',".format(current_version),
+        "version='{}',".format(next_version),
+    )
+    if not dry_run:
+        with open('jobs/setup.py', 'w') as f:
+            f.write(setup_py)
+
+    # Edit jobs/CHANGELOG.rst
+    with open('jobs/CHANGELOG.rst') as f:
+        original = f.read()
+    assert '\n\n'.join(change_log) not in original
+    new_change_log = original.replace(
+        '=========',
+        '=========\n\n{}\n\n'.format(
+            '\n\n'.join(change_log)
+        )
+    )
+    if not dry_run:
+        with open('jobs/CHANGELOG.rst', 'w') as f:
+            f.write(new_change_log)
+
+    # Actually commit this change.
+    commit_message = f'Bump {next_version}'
+    if dry_run:
+        print('git add jobs/CHANGELOG.rst jobs/setup.py')
+        print(
+            f'git commit -m "{commit_message}"'
+        )
+    else:
+        subprocess.check_call([
+            'git', 'add', 'jobs/CHANGELOG.rst', 'jobs/setup.py',
+        ])
+        subprocess.check_call([
+            'git', 'commit', '-m', commit_message,
+        ])
+
+    # Commit these changes
+    tag_name = tag_name_format.format(version=next_version)
+    tag_body = '\n\n'.join(['- {}'.format(x) for x in commit_messages])
+    if dry_run:
+        print(
+            f'git tag -s -a {tag_name} -m "...See CHANGELOG output above..."'
+        )
+    else:
+        subprocess.check_call([
+            'git',
+            'tag',
+            '-s',
+            '-a', tag_name,
+            '-m', tag_body,
+        ])
```

Comment:
Instead of having several places with `check_call` / `check_output` + decode, and a `if dry_run` switch, you could have a small helper : 

```py
output = run_command(f"git tag -s -a {tag_name} -m {tag_body}", dry_run=dry_run)\
```